From 0fa2893224fb6e864e96cfc33aec094cce579776 Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 May 2026 13:30:20 +0800
Subject: [PATCH] board: aspeed: add ast2705-dcscm board support

Add the DCSCM board files for the AST2705-based dual-channel server
chassis management board.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I1ce3f97795ec5249208497b82f0957bef31759c9
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 board/aspeed/ast2705_dcscm/Kconfig         | 17 +++++++++++++++++
 board/aspeed/ast2705_dcscm/Makefile        |  1 +
 board/aspeed/ast2705_dcscm/ast2705_dcscm.c | 10 ++++++++++
 3 files changed, 28 insertions(+)
 create mode 100644 board/aspeed/ast2705_dcscm/Kconfig
 create mode 100644 board/aspeed/ast2705_dcscm/Makefile
 create mode 100644 board/aspeed/ast2705_dcscm/ast2705_dcscm.c

diff --git a/board/aspeed/ast2705_dcscm/Kconfig b/board/aspeed/ast2705_dcscm/Kconfig
new file mode 100644
index 0000000000..3c2fa8d041
--- /dev/null
+++ b/board/aspeed/ast2705_dcscm/Kconfig
@@ -0,0 +1,17 @@
+if TARGET_AST2705_DCSCM
+
+config SYS_BOARD
+	default "ast2705_dcscm"
+
+config SYS_VENDOR
+	default "aspeed"
+
+config SYS_CONFIG_NAME
+	string "board configuration name"
+	default "ast2705_dcscm"
+
+config AVENUE_CITY_CRB
+	bool "Support for Intel Avenue City CRB"
+	default n
+
+endif
diff --git a/board/aspeed/ast2705_dcscm/Makefile b/board/aspeed/ast2705_dcscm/Makefile
new file mode 100644
index 0000000000..d85c931794
--- /dev/null
+++ b/board/aspeed/ast2705_dcscm/Makefile
@@ -0,0 +1 @@
+obj-y += ast2705_dcscm.o
diff --git a/board/aspeed/ast2705_dcscm/ast2705_dcscm.c b/board/aspeed/ast2705_dcscm/ast2705_dcscm.c
new file mode 100644
index 0000000000..296a2ecd9d
--- /dev/null
+++ b/board/aspeed/ast2705_dcscm/ast2705_dcscm.c
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) ASPEED Technology Inc.
+ */
+#include <asm/io.h>
+
+int board_early_init_f(void)
+{
+	return 0;
+}
-- 
2.43.0

